### PR TITLE
Contract creation on non-empty code or non-zero nonce throws an exception

### PR DESCRIPTION
**Pull request by pirapira**
This implements a protocol change which is very hard to hit anyway.  The idea is to avoid overwriting contracts even when the calculated address coincides with existing something.  Here "existing something" is an account with non-zero nonce or non-empty code.  This is consistent with EIP86 planned in Constantinople.

There have been some test failures happening only on the address-collision cases, but most probably the time spent there is wasted because keccak256 collisions cause far bigger problems anyway.  It makes sense to make the protocol specification simpler for the address collision case.

### CODE REVIEW

**Comment by pirapira** [comment]
This implements https://github.com/ethereum/EIPs/issues/684

**Comment by fulldecent** [comment]
Is there any more discussion on this. This is something I wondered too when reading the spec.

File: Paper.tex [hunk 1]
```diff
@@ -760,7 +760,7 @@ \section{Contract Creation}\label{ch:create}\hypertarget{endow}{}
 1 & \text{otherwise}
 \end{cases} \\
 \nonumber \text{where} \\
-F &\equiv \big((\boldsymbol{\sigma}^{**} = \varnothing \ \wedge\ \mathbf{o} = \varnothing) \vee\  g^{**} < c \ \vee\  |\mathbf{o}| > 24576\big)
+F \equiv &\big(\boldsymbol{\sigma}[a]_c \neq \texttt{\small KEC}\big(()\big) \ \vee\ \boldsymbol{\sigma}[a]_n \neq 0 \ \vee\ (\boldsymbol{\sigma}^{**} = \varnothing \ \wedge\ \mathbf{o} = \varnothing) \\\nonumber  & \vee\ g^{**} < c \ \vee\  |\mathbf{o}| > 24576\big)
```
Comment (livnev) [inline Paper.tex]: I believe this should be merged. I know that the yellow paper hasn't caught up to Constantinople yet but this would be a start.

**Comment by ytrezq** [comment]
@pirapira since this is now activated on mainnet, maybe the tag is no longer relavant., isn t it?

**Comment by yperbasis** [comment]
For reference: [geth implementation](https://github.com/ethereum/go-ethereum/blob/v1.10.3/core/vm/evm.go#L439)

**Review by yperbasis — APPROVED** [review]

**Pull request merged**